src/app/Providers/ProvidersPage.tsx

```diff
@@ -72,13 +72,13 @@ const ProvidersPage: React.FunctionComponent = () => {
 
   const activeProviderType = match?.params.providerType || null;
 
+  const isValidProviderType = !!(activeProviderType && PROVIDER_TYPES.includes(activeProviderType));
+
   React.useEffect(() => {
-    if (
-      (!activeProviderType && availableProviderTypes.length > 0) ||
-      (activeProviderType && !availableProviderTypes.includes(activeProviderType))
-    )
+    if ((!isValidProviderType || !activeProviderType) && availableProviderTypes.length > 0) {
```

Comment:
@seanforyou23 I thought of one more weird corner case here that I would be ok with not handling: If you go to the URL for e.g. `/providers/openshift` when there are no openshift providers, that is a valid provider type so it will still render the page (with a blank body). The tab won't exist, so users could only get there if they like bookmarked it or something and then removed a provider. If we want to handle this we should maybe render a real empty state on that page instead of a blank body in that case.

I think I'm really splitting hairs here though, so I'm totally fine leaving that alone.

Reply:
Ah, it's a good point! Yea that should be a nicely formatted page for sure. I don't mind doing that as part of this - I'll take a best guess at what the empty state should say. @vconzola any preferences?

Reply:
https://github.com/konveyor/forklift-ui/pull/729/commits/4ea3d7dd3bd1a6dca52c5a09c302cf3d3a458bd7

<img width="500" alt="Screen Shot 2021-08-02 at 9 42 16 AM" src="https://user-images.githubusercontent.com/5942899/127871408-79817125-0c8e-4568-9a44-8da509d8229a.png">